PyPi install guides improvements (#5756)
openvinotoolkit/openvino: docs/install_guides/pypi-openvino-dev.md

|------------------|---------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [Model Optimizer](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html) | `mo` |**Model Optimizer** imports, converts, and optimizes models that were trained in popular frameworks to a format usable by Intel tools, especially the Inference Engine. <br>Popular frameworks include Caffe\*, TensorFlow\*, MXNet\*, and ONNX\*.                                               |
| [Benchmark Tool](https://docs.openvinotoolkit.org/latest/openvino_inference_engine_tools_benchmark_tool_README.html)| `benchmark_app` | **Benchmark Application** allows you to estimate deep learning inference performance on supported devices for synchronous and asynchronous modes.                                              |

> | [Accuracy Checker](https://docs.openvinotoolkit.org/latest/omz_tools_accuracy_checker.html) and <br> [Annotation Converter](https://docs.openvinotoolkit.org/latest/omz_tools_accuracy_checker_annotation_converters.html) | `accuracy_check` <br> `convert_annotation` |**Accuracy Checker**  is a deep learning accuracy validation tool which allows you collect accuracy metrics against popular datasets. The main advantages of the tool are the flexibility of configuration and an impressive set of supported datasets, preprocessing, postprocessing, and metrics. <br> **Annotation Converter** is a utility for offline conversion dataset to suitable for metric evaluation format used in Accuracy Checker.                                            |

Reviewer: ```suggestion
| [Accuracy Checker](https://docs.openvinotoolkit.org/latest/omz_tools_accuracy_checker.html) and <br> [Annotation Converter](https://docs.openvinotoolkit.org/latest/omz_tools_accuracy_checker_annotation_converters.html) | `accuracy_check` <br> `convert_annotation` |**Accuracy Checker**  is a deep learning accuracy validation tool which allows you to collect accuracy metrics against popular datasets. The main advantages of the tool are the flexibility of configuration and an impressive set of supported datasets, preprocessing, postprocessing, and metrics. <br> **Annotation Converter** is a utility for offline conversion of datasets to the format suitable for metric evaluation that is used in Accuracy Checker.                                            |
```

> | [Post-Training Optimization Tool](https://docs.openvinotoolkit.org/latest/pot_README.html)| `pot` |**Post-Training Optimization Tool** allows you to optimize trained models with advanced capabilities, such as quantization and low-precision optimizations, without the need to re-train or fine-tune models.                                             |

Reviewer 2: re-train ---> retrain

> | [Model Downloader and other OMZ tools](https://docs.openvinotoolkit.org/latest/omz_tools_downloader.html)| `omz_downloader` <br> `omz_converter` <br> `omz_quantizer` <br> `omz_info_dumper`| **Model Downloader** is a tool for getting access to collection of high quality and extremely fast pre-trained deep learning [public](https://docs.openvinotoolkit.org/latest/omz_models_group_public.html) and [intel](https://docs.openvinotoolkit.org/latest/omz_models_group_intel.html)-trained models. These free pre-trained models can be used instead of training your own models to speed-up the development and production deployment process.  The principle of the tool is as follows: it downloads model files from online sources and, if necessary, patch them to make them more usable with Model Optimizer. A number of additional tools are also provided to automate the process of working with downloaded models:<br> **Model Converter** is a tool for conversion the models that are stored not in the Inference Engine IR format into that format using Model Optimizer. <br> **Model Quantizer** is a tool for automatic quantization full-precision models in the IR format into low-precision versions using Post-Training Optimization Tool. <br> **Model Information Dumper** is helper utility for dumping information about the models in a stable machine-readable format.|

Reviewer 3: ```suggestion
| [Model Downloader and other OMZ tools](https://docs.openvinotoolkit.org/latest/omz_tools_downloader.html)| `omz_downloader` <br> `omz_converter` <br> `omz_quantizer` <br> `omz_info_dumper`| **Model Downloader** is a tool for getting access to collection of high quality and extremely fast pre-trained deep learning [public](https://docs.openvinotoolkit.org/latest/omz_models_group_public.html) and [intel](https://docs.openvinotoolkit.org/latest/omz_models_group_intel.html)-trained models. These free pre-trained models can be used instead of training your own models to speed-up the development and production deployment process.  The principle of the tool is as follows: it downloads model files from online sources and, if necessary, patches them to make them more usable with Model Optimizer. A number of additional tools are also provided to automate the process of working with downloaded models:<br> **Model Converter** is a tool for converting the models that are stored not in the Inference Engine IR format into that format using Model Optimizer. <br> **Model Quantizer** is a tool for automatic quantization full-precision models in the IR format into low-precision versions using Post-Training Optimization Tool. <br> **Model Information Dumper** is a helper utility for dumping information about the models in a stable machine-readable format.|
```

> | [Model Downloader and other OMZ tools](https://docs.openvinotoolkit.org/latest/omz_tools_downloader.html)| `omz_downloader` <br> `omz_converter` <br> `omz_quantizer` <br> `omz_info_dumper`| **Model Downloader** is a tool for getting access to collection of high quality and extremely fast pre-trained deep learning [public](https://docs.openvinotoolkit.org/latest/omz_models_group_public.html) and [intel](https://docs.openvinotoolkit.org/latest/omz_models_group_intel.html)-trained models. These free pre-trained models can be used instead of training your own models to speed-up the development and production deployment process.  The principle of the tool is as follows: it downloads model files from online sources and, if necessary, patch them to make them more usable with Model Optimizer. A number of additional tools are also provided to automate the process of working with downloaded models:<br> **Model Converter** is a tool for conversion the models that are stored not in the Inference Engine IR format into that format using Model Optimizer. <br> **Model Quantizer** is a tool for automatic quantization full-precision models in the IR format into low-precision versions using Post-Training Optimization Tool. <br> **Model Information Dumper** is helper utility for dumping information about the models in a stable machine-readable format.|

Reviewer: ```suggestion
| [Model Downloader and other OMZ tools](https://docs.openvinotoolkit.org/latest/omz_tools_downloader.html)| `omz_downloader` <br> `omz_converter` <br> `omz_quantizer` <br> `omz_info_dumper`| **Model Downloader** is a tool for getting access to collection of high quality and extremely fast pre-trained deep learning [public](https://docs.openvinotoolkit.org/latest/omz_models_group_public.html) and [Intel](https://docs.openvinotoolkit.org/latest/omz_models_group_intel.html)-trained models. These free pre-trained models can be used instead of training your own models to speed up the development and production deployment process.  The principle of the tool is as follows: it downloads model files from online sources and, if necessary, patches them to make them more usable with Model Optimizer. A number of additional tools are also provided to automate the process of working with downloaded models:<br> **Model Converter** is a tool for conversion of the models that are stored in a format other than the Inference Engine IR format into that format using Model Optimizer. <br> **Model Quantizer** is a tool for automatic quantization of full-precision models in the IR format into low-precision versions using Post-Training Optimization Tool. <br> **Model Information Dumper** is a helper utility for dumping information about the models in a stable machine-readable format.|
```

| [Benchmark Tool](https://docs.openvinotoolkit.org/latest/openvino_inference_engine_tools_benchmark_tool_README.html)| `benchmark_app` | **Benchmark Application** allows you to estimate deep learning inference performance on supported devices for synchronous and asynchronous modes.                                              |
| [Accuracy Checker](https://docs.openvinotoolkit.org/latest/omz_tools_accuracy_checker.html) and <br> [Annotation Converter](https://docs.openvinotoolkit.org/latest/omz_tools_accuracy_checker_annotation_converters.html) | `accuracy_check` <br> `convert_annotation` |**Accuracy Checker**  is a deep learning accuracy validation tool that allows you to collect accuracy metrics against popular datasets. The main advantages of the tool are the flexibility of configuration and an impressive set of supported datasets, preprocessing, postprocessing, and metrics. <br> **Annotation Converter** is a utility for offline conversion of datasets to the format suitable for metric evaluation used in Accuracy Checker.                                            |

> | [Post-Training Optimization Tool](https://docs.openvinotoolkit.org/latest/pot_README.html)| `pot` |**Post-Training Optimization Tool** allows you to optimize trained models with advanced capabilities, such as quantization and low-precision optimizations, without the need to retrain or fine-tune models.                                             |

Reviewer 4: POT can be used (and actually quite often used) as a library or package through its API. So it is worth mentioning how to reach this API. 
Right now it is located in the `compression.api`. In the next release, we are going to rename it to `pot.api`. We cannot do it in the LTS. @MaximProshin, can comment more why it is so.

Author: @helena-intel , @ryanloney, @andrew-zaytsev  Should we  add description for component's module names and API, as Alex proposed? IMHO, it is rather developer guide info than install guide.

Reviewer 5: @MaximProshin @AlexKoff88  why can't we change this in an LTS release? The goal of LTS should be to provide stable API experience for runtime. POT is a developer tool. If it is not impacting a production deployment, I don't see why this cannot be changed now.

Author: Added reference to POT API.

> 

Reviewer 5: Please say "Open Model Zoo tools" instead of "OMZ tools" for those seeing the acronym the first time

### Step 4. Install the Package

> To install and configure components of the development package for working with specific frameworks, use the `pip install openvino-dev[extras]` command, where `extras` is a list of extras from the table below: 

Reviewer 2: To install and configure components ---> To install and configure the components

> | DL Framework                                                                     | Extras                          |

Reviewer: ```suggestion
| DL Framework                                                                     | Extra                          |
```

(there is only one extra per framework)

| :------------------------------------------------------------------------------- | :-------------------------------|

> |   [TensorFlow 1.x*](https://www.tensorflow.org/versions#tensorflow_1)            | tensorflow                      |
> |   [TensorFlow 2.x*](https://www.tensorflow.org/versions#tensorflow_2)            | tensorflow2                     |

Reviewer: ```suggestion
|   [TensorFlow* 1.x](https://www.tensorflow.org/versions#tensorflow_1)            | tensorflow                      |
|   [TensorFlow* 2.x](https://www.tensorflow.org/versions#tensorflow_2)            | tensorflow2                     |
```

|   [ONNX*](https://github.com/microsoft/onnxruntime/)                             | onnx                            |
|   [Caffe*](https://caffe.berkeleyvision.org/)                                    | caffe                           |

> |   [Caffe 2*](https://caffe2.ai/)                                                 | caffe2                          |

Reviewer: ```suggestion
|   [Caffe2*](https://caffe2.ai/)                                                 | caffe2                          |
```

Author: fixed

|   [MXNet*](https://mxnet.apache.org/)                                            | mxnet                           |
|   [PyTorch*](https://pytorch.org/)                                               | pytorch                         |

> |   [Kaldi*](https://kaldi-asr.org/)                                               | kaldi                           |

Reviewer: I would suggest listing these in alphabetical order.

> For example, to install and configure components for working with TensorFlow 2.x, MXNet and Caffe, use the folowing command:  

Reviewer 2: configure components --> configure the components
use the folowing command --> use the following command

   ```sh
   pip install openvino-dev[tensorflow2,mxnet,caffe]
   ``` 

> ### Step 5. Verify that the Package is Installed

Reviewer 2: Verify that the Package is Installed ---> Verify that the Package Is Installed

Run the command below (this may take a few seconds):
```sh

> mo -h

Reviewer 3: When I installed openvino-dev 2021.4 on Ubuntu 20.4 with Python 3.7 `mo -h` worked: it showed the help file as expected. But `from openvino.inference_engine import IECore` failed, because I had not installed libpython3.7-dev. Is there another command that can be used here that would catch that error?

Author: Please find an answer in Troubleshooting section of this document.

Reviewer 3: Sadly that doesn't work on Ubuntu 20.  It is not trivial to install Python 3.7 libraries there. Much easier to just use Python 3.8. But also, it is nice if the verification command verifies that the installation indeed worked, and OpenVINO works as expected.

Author: 1. Python default version on Ubuntu 18.04 and Ubuntu 20.04
Python 3.6 is default version on [Ubuntu 18.04](https://packages.ubuntu.com/bionic/allpackages) with ability to install  Python 3.7 from Ubuntu repository
Python 3.8 is default version on [Ubuntu 20.04](https://packages.ubuntu.com/focal/allpackages) and Python 3.7 is absent.
So I clarified the Ubuntu version in troubleshooting instructions.
2. Added additional verification step for mentioned issue (the same as defined for runtime package).
